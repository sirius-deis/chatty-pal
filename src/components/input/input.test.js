import { render, screen, fireEvent } from "@testing-library/react";
import Input from "./input";

describe("Input component", () => {
  const fn = jest.fn();

  beforeEach(() => {
    fn.mockClear();
  });
  
  it("should match snapshot", () => {
    const { container } = render(
      <Input type="text" placeholder="Enter text" />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render input with provided type and placeholder", () => {
    render(<Input type="text" placeholder="Enter text" />);
    expect(screen.getByPlaceholderText(/Enter text/)).toBeInTheDocument();
  });
  it("should change value", () => {
    render(<Input type="text" placeholder="Enter text" onChange={fn} />);
    fireEvent.change(screen.getByPlaceholderText(/Enter text/), {
      target: { value: "New value" },
    });
    expect(screen.getByPlaceholderText(/Enter text/)).toHaveValue("New value");
  });
  it("should call onChange function", () => {
    render(<Input type="text" placeholder="Enter text" onChange={fn} name="name" />);
    fireEvent.change(screen.getByPlaceholderText(/Enter text/), {
      target: { value: "New value" },
    });
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith("New value", "name");
  });
});
