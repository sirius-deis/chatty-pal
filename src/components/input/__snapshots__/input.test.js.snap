// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should match snapshot 1`] = `
<div>
  <input
    class="sc-aXZVg fPdPqO"
    placeholder="Enter text"
    type="text"
    value=""
  />
</div>
`;
